libethereum/Executive.cpp

```diff
@@ -365,7 +365,7 @@ bool Executive::createOpcode(Address const& _sender, u256 const& _endowment, u25
 
 bool Executive::create2Opcode(Address const& _sender, u256 const& _endowment, u256 const& _gasPrice, u256 const& _gas, bytesConstRef _init, Address const& _origin, u256 const& _salt)
 {
-    m_newAddress = right160(sha3(_sender.asBytes() + toBigEndian(_salt) + sha3(_init).asBytes()));
+    m_newAddress = right160(sha3(_sender.asBytes() + toBigEndian(_salt) + _init));
```

Comment:
I think the consensus is going towards adding the `0xff` prefix. The benefit is that old addresses can't clash with new ones, which is a win because it prevents a class of corner cases that might appear in old create code.